### PR TITLE
docs: update 9.0 upgrade instructions

### PR DESCRIPTION
**Pull request by daniellee**
This takes the upgrade instructions from the What's new for 9.0 and adds them to the upgrade instructions.

### CODE REVIEW

File: docs/sources/setup-grafana/upgrade-grafana.md [hunk 1]
```diff
@@ -405,6 +405,91 @@ This behavior was not very intuitive and creates issues for users who want to ch
...
+
+- Built-in roles are now called basic roles. They now consist of permissions, not roles.
+- The Terraform `builtin_role_assignment` resource is deprecated. Please use [grafana_role](https://registry.terraform.io/providers/grafana/grafana/latest/docs/resources/role) resource instead.
+- Grafana provisioning has a new schema. Please refer to [Provisioning RBAC with Grafana](https://grafana.com/docs/grafana/latest/administration/roles-and-permissions/access-control/rbac-grafana-provisioning/) to learn more about provisioning.
+- Basic roles no longer support permission inheritance. Previously, when permissions of a Viewer basic role were modified, it was propagated to the Editor and Admin basic roles. With the Grafana 9.0 release, this is not the case anymore.
+- Several role-based access control actions have been renamed. All database entries that use legacy action names will be migrated to use the new names, but provisioning files and scripts will have to be updated by the user. This change also means that if Grafana is downgraded from 9.0 to a lower version, some role-based access control permissions will not be resolved correctly.
```
Comment (vtorosyan) [inline docs/sources/setup-grafana/upgrade-grafana.md:418]: @grafana/grafana-authnz-team on this one, I am thinking what else we can do to support the upgrade path better, particularly with the upgrade-downgrade scenario.

Would it make sense, for example, to add the SQL queries which reset migrations to a public gist and reference it from here for users to use? Or any other ideas?

File: docs/sources/setup-grafana/upgrade-grafana.md [hunk 2]
```diff
@@ -405,6 +405,91 @@ This behavior was not very intuitive and creates issues for users who want to ch
...
 Existing dashboards that still have panels with a `datasource` set to null will be migrated when the dashboard opens. The migration will set the data source property to the **current** default data source.
 
+## Upgrading to 9.0
+
+### Role-based access control: changes for general release
+
+Fine-grained access control is now called "Role-based access control (RBAC)". As part of the Grafana 9.0 release, the feature is generally available, and there are several breaking changes:
+
+- Built-in roles are now called basic roles. They now consist of permissions, not roles.
+- The Terraform `builtin_role_assignment` resource is deprecated. Please use [grafana_role](https://registry.terraform.io/providers/grafana/grafana/latest/docs/resources/role) resource instead.
```
Comment (IevaVasiljeva) [inline docs/sources/setup-grafana/upgrade-grafana.md:415]: This is actually not a very smooth experience, I've documented the steps that you need to take here: https://github.com/grafana/grafana-enterprise/issues/4108

I think we should also include that somewhere in the docs (not as part of this PR obviously, just wanted to call it out).

File: docs/sources/setup-grafana/upgrade-grafana.md (hunk 1 above)
Comment (IevaVasiljeva) [reply docs/sources/setup-grafana/upgrade-grafana.md:418]: I think that would be fine, as long as we make it very clear what exactly the script does.

**Comment by vtorosyan** [comment]
I opened the change for review, and went over the docs, don't have full context but I think content is mergeable.

**Review by chri2547 — APPROVED** [review]

**Pull request merged**